proposals/user-defined-compound-assignment.md

```diff
@@ -0,0 +1,568 @@
+    : unary_operator_declarator
+    | binary_operator_declarator
+    | conversion_operator_declarator
++   | increment_operator_declarator
++   | compound_assignment_operator_declarator
+    ;
+
+unary_operator_declarator
+    : type 'operator' overloadable_unary_operator '(' fixed_parameter ')'
+    ;
+
+logical_negation_operator
+    : '!'
+    ;
+
+overloadable_unary_operator
+-   : '+' | '-' | logical_negation_operator | '~' | '++' | '--' | 'true' | 'false'
++   : '+' | '-' | logical_negation_operator | '~' | 'true' | 'false'
+    ;
+
+binary_operator_declarator
+    : type 'operator' overloadable_binary_operator
+        '(' fixed_parameter ',' fixed_parameter ')'
+    ;
+
+overloadable_binary_operator
+    : '+'  | '-'  | '*'  | '/'  | '%'  | '&' | '|' | '^'  | '<<' 
+    | right_shift | '==' | '!=' | '>' | '<' | '>=' | '<='
+    ;
+
+conversion_operator_declarator
+    : 'implicit' 'operator' type '(' fixed_parameter ')'
+    | 'explicit' 'operator' type '(' fixed_parameter ')'
+    ;
+
++increment_operator_declarator
++   : type 'operator' overloadable_increment_operator '(' fixed_parameter ')'
++   | 'void' 'operator' overloadable_increment_operator '(' ')'
++   ;
+
++overloadable_increment_operator
++   : '++' | '--'
++    ;
+
++compound_assignment_operator_declarator
++   : 'void' 'operator' overloadable_compound_assignment_operator
++       '(' fixed_parameter ')'
++   ;
+
++overloadable_compound_assignment_operator
++   : '+=' | '-=' | '*=' | '/=' | '%=' | '&=' | '|=' | '^=' | '<<='
++   | right_shift_assignment
++   | unsigned_right_shift_assignment
++   ;
+
+operator_body
+    : block
+    | '=>' expression ';'
+    | ';'
+    ;
+```
+
+There are five categories of overloadable operators: [unary operators](#unary-operators), [binary operators](#binary-operators),
+[conversion operators](#conversion-operators), [increment operators](#increment-operators), [compound assignment operators](#compound-assignment-operators).
+
+>The following rules apply to all operator declarations:
+>
+>- An operator declaration shall include ~~both~~ a `public` ~~and a `static`~~ modifier.
+
+Compound assignment and instance increment operators can hide operators declared in a base class. Therefore,
+the following paragraph is no longer accurate and should either be adjusted accordingly, or it can be removed: 
+>Because operator declarations always require the class or struct in which the operator is declared to participate in the signature of the operator,
+it is not possible for an operator declared in a derived class to hide an operator declared in a base class. Thus, the `new` modifier is never
+required, and therefore never permitted, in an operator declaration.
+
+### Unary operators
+[unary-operators]: #unary-operators
+
+See https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/classes.md#15102-unary-operators.
+
+An operator declaration shall include a `static` modifier and shall not include an `override` modifier.
```

Comment:
Should the `new` (and `sealed` if we want to support it) modifier be also forbidden for unary, binary, conversion and static increment operators?

Reply:
> Should the `new` (and `sealed` if we want to support it) modifier be also forbidden for unary, binary, conversion and static increment operators?

I do not see any good reason to forbid `new`, `sealed` is already allowed by the grammar.